src/address.rs

```diff
@@ -269,23 +291,33 @@ impl<A, Rc: RefCounter> Eq for Address<A, Rc> {}
 /// address will compare as greater than a weak one.
 impl<A, Rc: RefCounter, Rc2: RefCounter> PartialOrd<Address<A, Rc2>> for Address<A, Rc> {
     fn partial_cmp(&self, other: &Address<A, Rc2>) -> Option<Ordering> {
-        Some(match self.0.inner_ptr().cmp(&other.0.inner_ptr()) {
-            Ordering::Equal => self.0.is_strong().cmp(&other.0.is_strong()),
-            ord => ord,
-        })
+        Some(
+            match Arc::as_ptr(&self.inner).cmp(&Arc::as_ptr(&other.inner)) {
```

Comment:
Just to check, does this properly compare data ptrs and not vtables?

Reply:
According to the docs, yes: https://doc.rust-lang.org/std/sync/struct.Arc.html#method.ptr_eq

Reply:
Interesting, I guess this also needs to be changed then.